src/fondant/component/data_io.py

```diff
@@ -204,7 +207,34 @@ def _write_dataframe(self, dataframe: dd.DataFrame) -> dd.core.Scalar:
             field.name: field.type.value
             for field in self.operation_spec.outer_produces.values()
         }
-        return self._create_write_task(dataframe, location=location, schema=schema)
+
+        # The id needs to be added explicitly since we will convert this to a PyArrow schema
+        # later and use it in the `pandas.to_parquet` method.
+        schema.update(
+            {
+                "id": pa.string(),
+            },
+        )
+
+        # Convert to delayed since computing a dataframe tries to return the complete result,
+        # keeping references to all completed tasks, and preventing release of memory.
+        # https://distributed.dask.org/en/stable/memory.html#difference-with-dask-compute
+        # https://dask.discourse.group/t/improving-pipeline-resilience-when-using-to-parquet-and-preemptible-workers/2141
+        to_parquet_tasks = [
+            d.to_parquet(
+                os.path.join(location, f"part.{i}.parquet"),
+                schema=pa.schema(list(schema.items())),
+            )
+            for (i, d) in enumerate(dataframe.to_delayed())
+        ]
+
+        client: dask.distributed.Client = dask.distributed.get_client()
```

Comment:
`get_client()` will retrieve the global client I guess. But shouldn't we use the client we have initialised in the setup method explicitly to make clear where the client comes from?

Reply:
In our setup there should only be one client, and `get_client` retrieves the client connected to the current worker. So this should provide the correct one.